### PR TITLE
feat(Avatar): add 2xs Avatar size

### PR DESCRIPTION
**Pull request by PahaN47**

### CODE REVIEW

**Comment by amje** [comment]
@PahaN47 Please ensure new size is correctly displayed for Avatar related components: `AvatarStack`, `User`, `UserLabel`

**Comment by DakEnviy** [comment]
Let's add `2xs` size for `User` component. Also, I believe that 12px font-size would be better in `User` component for sizes `xs` and `2xs`

File: src/components/User/User.scss [hunk 1]
```diff
@@ -36,10 +42,15 @@ $block: '.#{variables.$ns}user';
...
     &_size {
-        &_xs {
+        &_xs,
+        &_2xs {
             #{$block}__avatar + #{$block}__info {
                 margin-inline-start: 6px;
             }
+
+            #{$block}__info {
+                @include user-text-small();
```
[the remark this answers is not in the document]
Comment (amje) [reply src/components/User/User.scss:52]: Maybe we shouldn't allow `2xs` in User and UserLabel components

File: src/components/User/User.scss [hunk 2]
```diff
@@ -3,6 +3,12 @@
 
 $block: '.#{variables.$ns}user';
 
+@mixin user-text-small() {
+    font-size: var(--g-text-code-inline-1-font-size);
+    line-height: var(--g-text-code-inline-1-line-height);
+    font-weight: var(--g-text-body-font-weight);
```
Comment (DakEnviy) [inline src/components/User/User.scss:9]: Let's add CSS API for font styles

File: src/components/User/README.md [hunk 3]
```diff
@@ -59,9 +61,18 @@ LANDING_BLOCK-->
...
 | qa              | HTML `data-qa` attribute, used in tests |                              `string`                              |         |
+
+## CSS API
+
+| Name                         | Description                                                |
+| :--------------------------- | :--------------------------------------------------------- |
+| `--g-user-font-size`         | Name and description font size for `size='s'` and larger   |
+| `--g-user-line-height`       | Name and description line height for `size='s'` and larger |
+| `--g-user-small-font-size`   | Name font size for `size='xs'` and smaller                 |
+| `--g-user-small-line-height` | Name line height for `size='xs'` and smaller               |
```
Comment (amje) [inline src/components/User/README.md:78]: CSS API is designed to ignore all props, so only 2 vars are relevant: `--g-user-font-size`, `--g-user-line-height`. There should not be CSS API for props variants

File: src/components/User/constants.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import type {UserSize} from './types';
+
+export const NO_DESCRIPTION_SIZES: UserSize[] = ['xs', '2xs'];
+export const DEFAULT_USER_SIZE: UserSize = 'm';
```
Comment (amje) [inline src/components/User/constants.ts:4]:
```suggestion
export const DEFAULT_SIZE: UserSize = 'm';
```

File: src/components/User/constants.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import type {UserSize} from './types';
+
+export const NO_DESCRIPTION_SIZES: UserSize[] = ['xs', '2xs'];
```
Comment (amje) [inline src/components/User/constants.ts:3]:
```suggestion
export const COMPACT_SIZES: UserSize[] = ['xs', '2xs'];
```

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Review by DakEnviy — APPROVED** [review]

**Pull request merged**